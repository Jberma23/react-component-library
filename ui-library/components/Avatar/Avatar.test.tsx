import { render } from "@testing-library/react";
import { Avatar } from ".";
import { axe, toHaveNoViolations } from 'jest-axe';

describe(Avatar, () => {
  it("renders with required props", () => {
    const { asFragment } = render(<Avatar>Hello World</Avatar>);
    expect(asFragment()).toMatchSnapshot();
  });
  it("passes all a11y checks", async () => {
    const { container } = render(<Avatar>Hello World</Avatar>);
    expect(await axe(container)).toHaveNoViolations();
  });
});
