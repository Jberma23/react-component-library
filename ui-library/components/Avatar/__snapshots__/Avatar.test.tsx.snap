// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders with required props 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center w-[40px] h-[40px] p-s rounded-full border-4 bg-primary-200 border-solid border-white hover:border-gray-300 active:border-primary-300 duration-default transition-colors"
  >
    Hello World
  </div>
</DocumentFragment>
`;
